pkg/controller/certificates/keymanager/keymanager_controller.go

```diff
@@ -319,7 +323,10 @@ func (c *controller) updateOrApplyStatus(ctx context.Context, crt *cmapi.Certifi
 			Status:     cmapi.CertificateStatus{NextPrivateKeySecretName: crt.Status.NextPrivateKeySecretName},
 		})
 	} else {
-		_, err := c.client.CertmanagerV1().Certificates(crt.Namespace).UpdateStatus(ctx, crt, metav1.UpdateOptions{})
+		_, err := c.client.CertmanagerV1().Certificates(crt.Namespace).UpdateStatus(ctx, &cmapi.Certificate{
+			ObjectMeta: crt.ObjectMeta,
+			Status:     crt.Status,
+		}, metav1.UpdateOptions{})
 		return err
```

Comment:
Without this change, the defaults, which are now applied early in the ProcessItem function, caused the unit-tests to fail, because the changes made to the certificate spec were being recorded by the fake client, despite this being an UpdateStatus call.

I think this is more correct anyway and more closely mirrors what we do for serversideapply in the code above here.

Here's how the tests failed:
 * https://prow.infra.cert-manager.io/view/gs/cert-manager-prow-artifacts/pr-logs/pull/cert-manager_cert-manager/7723/pull-cert-manager-master-make-test/1920123874280738816